src/components/FormEditEvent.js

```diff
@@ -0,0 +1,87 @@
+var React = require('react');
+var EventActions = require('../actions/EventActions');
+
+/**
+ * <FormEditEvent event={event} />
+ *  Update title, date, and notes
+ */
+var FormEditEvent = React.createClass({
+  propTypes: {
+    event: React.PropTypes.object.isRequired
+  },
+
+  componentWillMount() {
+    this.setStateFromEvent(this.props.event);
+  },
+
+  componentWillReceiveProps(nextProps) {
+    this.setStateFromEvent(nextProps.event);
+  },
+
+  setStateFromEvent(event) {
+    this.setState({
+      title: event.title,
+      dateArgs: event.moment.format("YYYY-MM-DD"),
+      notes: event.notes || ""
+    })
+  },
+
+  render() {
+    return (
+      <div>
+        <input
+          ref="eventTitle"
+          onChange={this.onTitleChanged}
+          onSubmit={this.updateEvent}
+          value={this.state.title}
+          type="text"
+          className="form-control input-md"
+          />
+        <input
+          type="date"
+          value={this.state.dateArgs}
+          onChange={this.onDateChanged}
+          className="form-control input-md"
+          />
+        <textarea
+          value={this.state.notes}
+          onChange={this.onNotesChanged}
+          className="form-control input-md"
+          rows="4"
+          />
+        <button
+          onClick={this.updateEvent}
+          className="btn btn-md btn-block btn-primary">
+          Save
+        </button>
+      </div>
+    );
+  },
+
+  onTitleChanged(e) {
+    this.setState({title: e.target.value});
+  },
+
+  onDateChanged(e) {
+    this.setState({dateArgs: e.target.value});
+  },
+
+  onNotesChanged(e) {
+    this.setState({notes: e.target.value});
+  },
```

Comment:
These `on*Change` methods are all pretty much the same. I like to keep things DRY with a more generic handler like...

```javascript
onFieldChange(fieldName, e) {
  this.setState({
    [fieldName]: e.target.value,
  });
},
```

...that I can partially apply for different fields:

```javascript
<input
  onChange={this.onFieldChange.bind(null, 'dateArgs')}
  type="date"
  value={this.state.dateArgs}
/>
<textarea
  onChange={this.onFieldChange.bind(null, 'notes')}
  value={this.state.notes}
/>
```